src/components/common/LangDropdown.jsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import lang from "../images/lang.png";
+import "../index.css";
+
+function LangDropdown() {
+  const { t, i18n } = useTranslation();
+
+  const handleChange = (e) => {
+    switch (e.target.innerText) {
+      case "Turkish":
+      case "Türkçe":
+      case "التركية":
+        i18n.changeLanguage("tr");
+        return;
+      case "Arabic":
+      case "Arapça":
+      case "العربية":
+        i18n.changeLanguage("ar");
+        return;
+      case "English":
+      case "İngilizce":
+      case "الإنجليزية":
+        i18n.changeLanguage("en");
+        return;
+      default:
+        return;
+    }
+  };
+
+  return (
+    <div className="langDropdown">
+      <img className="lang" alt="lang. selector" src={lang} />
+      <ul className="lang">
+        <li className="lang" onClick={(e) => handleChange(e)}>
```

Comment:
is there a reason you're not using the bootstrap dropdown component?
https://getbootstrap.com/docs/4.0/components/dropdowns/